### PR TITLE
grass 7.0.0

### PR DESCRIPTION
**Pull request by dunn**

### CODE REVIEW

File: Library/Formula/grass.rb [hunk 1]
```diff
@@ -1,42 +1,34 @@
...
-
-    patch do
-      url "https://gist.githubusercontent.com/jctull/0fe3db92a3e7c19fa6e0/raw/42e819f0a9b144de782c94f730dbc4da136e9227/grassPatchHead.diff"
-      sha1 "ffbe31682d8a7605d5548cdafd536f1c785d3a23"
-    end
+  # prevent installer trying to write to /Library
+  patch do
+    url "https://gist.githubusercontent.com/dunn/776ccb55d8d4535798bf/raw/288f43b37f7b1016efcefaea37ef5133d93c75b0/grass.diff"
+    sha256 "c7020dd282e0144bd900d0ac5d51ada486c071d3826db5405d2b6ef63ee2d7fc"
   end
```
Comment (mikemcquaid) [inline Library/Formula/grass.rb]: Can you please submit this patch to the upstream developers of this project and add a link to the upstream patch submission and explanation of why the patch is needed in a comment in the formula file. Thanks!

File: Library/Formula/grass.rb [hunk 2]
```diff
@@ -1,42 +1,34 @@
...
+  depends_on "readline"
   depends_on "unixodbc"
-  depends_on "fftw"
   depends_on "wxpython" => :recommended # prefer over OS X's version because of 64bit
-  depends_on :postgresql => :optional
   depends_on :mysql => :optional
-  depends_on "cairo"
-  depends_on "freetype"
+  depends_on :postgresql => :optional
   depends_on :x11  # needs to find at least X11/include/GL/gl.h
```
Comment (dunn) [inline Library/Formula/grass.rb]: Oh, hang on.  Should this whole formula be whisked off into `homebrew/x11`?

Comment (mikemcquaid) [inline Library/Formula/grass.rb]: Yes, maybe, if it can't be coaxed to work without.

**Comment by dunn** [comment]
OK, version 7 doesn't require X11 anymore, but there's a lot more different than I realized and it's a bit beyond me.  If anyone's interested though:

- New [requirements](http://svn.osgeo.org/grass/grass/trunk/REQUIREMENTS.html) and [install instructions](http://svn.osgeo.org/grass/grass/trunk/INSTALL)
- `MACOSX_SUPPDIR` needs to be set during `make install` to somewhere other than the default `/Library/GRASS`, when building the .app version
- The .app version currently requires [32-bit Python and 32-bit wxpython](http://grasswiki.osgeo.org/wiki/MacOSX_GRASS_errors)

**Comment by mikemcquaid** [comment]
Given this seems to have built OK I think I'm alright including it as-is if we remove the `:x11` dependency.

**Comment by dunn** [comment]
I tried actually using it and there's a Python dependency issue that causes the GUI to crash.  Set up a database and start a session, then:

```
Launching <wxpython> GUI in the background, please wait...
GRASS 7.0.0 (cats):/usr/local > wxnviz.py: This module requires the NumPy module, which could not be imported. It probably is not installed (it's not part of the standard Python distribution). See the Numeric Python site (http://numpy.scipy.org) for information on downloading source or binaries.
NumPy not found.

    This module requires the NumPy module, which could not be
    imported.  It probably is not installed (it's not part of the
    standard Python distribution). See the Numeric Python site
    (http://numpy.scipy.org) for information on downloading source or
    binaries.
NumPy not found.

    This module requires the NumPy module, which could not be
    imported.  It probably is not installed (it's not part of the
    standard Python distribution). See the Numeric Python site
    (http://numpy.scipy.org) for information on downloading source or
    binaries.
NumPy not found.

    This module requires the NumPy module, which could not be
    imported.  It probably is not installed (it's not part of the
    standard Python distribution). See the Numeric Python site
    (http://numpy.scipy.org) for information on downloading source or
    binaries.
wxplot.py: This module requires the NumPy module, which could not be imported. It probably is not installed (it's not part of the standard Python distribution). See the Numeric Python site (http://numpy.scipy.org) for information on downloading source or binaries.
NumPy not found.

    This module requires the NumPy module, which could not be
    imported.  It probably is not installed (it's not part of the
    standard Python distribution). See the Numeric Python site
    (http://numpy.scipy.org) for information on downloading source or
    binaries.
Traceback (most recent call last):
  File "/usr/local/Cellar/grass/7.0.0/grass-7.0.0/gui/wxpython/wxgui.py", line 142, in <module>
    sys.exit(main())
  File "/usr/local/Cellar/grass/7.0.0/grass-7.0.0/gui/wxpython/wxgui.py", line 134, in main
    app = GMApp(workspaceFile)
  File "/usr/local/Cellar/grass/7.0.0/grass-7.0.0/gui/wxpython/wxgui.py", line 49, in __init__
    wx.App.__init__(self, False)
  File "/usr/local/Cellar/python/2.7.9/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/wx-3.0-osx_cocoa/wx/_core.py", line 8628, in __init__
    self._BootstrapApp()
  File "/usr/local/Cellar/python/2.7.9/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/wx-3.0-osx_cocoa/wx/_core.py", line 8196, in _BootstrapApp
    return _core_.PyApp__BootstrapApp(*args, **kwargs)
  File "/usr/local/Cellar/grass/7.0.0/grass-7.0.0/gui/wxpython/wxgui.py", line 83, in OnInit
    workspace = self.workspaceFile)
  File "/usr/local/Cellar/grass/7.0.0/grass-7.0.0/gui/wxpython/lmgr/frame.py", line 119, in __init__
    self._menuTreeBuilder = LayerManagerMenuData()
  File "/usr/local/Cellar/grass/7.0.0/grass-7.0.0/gui/wxpython/lmgr/menudata.py", line 40, in __init__
    MenuTreeModelBuilder.__init__(self, filename, expandAddons=expandAddons)
  File "/usr/local/Cellar/grass/7.0.0/grass-7.0.0/gui/wxpython/core/menutree.py", line 64, in __init__
    xmlTree = etree.parse(filename)
  File "/usr/local/Cellar/python/2.7.9/Frameworks/Python.framework/Versions/2.7/lib/python2.7/xml/etree/ElementTree.py", line 1182, in parse
    tree.parse(source, parser)
  File "/usr/local/Cellar/python/2.7.9/Frameworks/Python.framework/Versions/2.7/lib/python2.7/xml/etree/ElementTree.py", line 657, in parse
    self._root = parser.close()
  File "/usr/local/Cellar/python/2.7.9/Frameworks/Python.framework/Versions/2.7/lib/python2.7/xml/etree/ElementTree.py", line 1654, in close
    self._raiseerror(v)
  File "/usr/local/Cellar/python/2.7.9/Frameworks/Python.framework/Versions/2.7/lib/python2.7/xml/etree/ElementTree.py", line 1506, in _raiseerror
    raise err
xml.etree.ElementTree.ParseError: no element found: line 1, column 0
```

**Comment by DomT4** [comment]
Looks like this needs a `numpy` dep vendored. We can do that into `libexec`, [Binwalk](https://github.com/Homebrew/homebrew/blob/master/Library/Formula/binwalk.rb) is an example.

**Comment by bfontaine** [comment]
Any update here?

**Comment by dunn** [comment]
The python stuff confused me, I was going to wait and see if #36810 makes it any easier.

**Comment by dunn** [comment]
Maybe we should let OSGeo officially take over this formula: https://github.com/OSGeo/homebrew-osgeo4mac/blob/master/Formula/grass-70.rb

**Comment by mikemcquaid** [comment]
> Maybe we should let OSGeo officially take over this formula: https://github.com/OSGeo/homebrew-osgeo4mac/blob/master/Formula/grass-70.rb

@dunn Makes sense but I'd like to see it move to the Homebrew organisation first.

**Comment by tdsmith** [comment]
I second handing this to OSGeo; can we bring them in yet?

**Comment by mikemcquaid** [comment]
@tdsmith @dakcarto has permissions to do so but hasn't yet.

**Comment by dunn** [comment]
Given that there is a 7.0 formula in the OSGeo tap, and given that there doesn't seem to be a ton of demand, I'm just going to close this; when OSGeo migrates to Homebrew/osgeo they'll probably use their existing formula anyway.

**Comment by dakcarto** [comment]
Hi. I have updated the `grass-70` in our tap, but it (like others) are now working with a GDAL/OGR v2.x formula dep. So, some work on getting v2.x working at Homebrew would be an ideal first step (like https://github.com/Homebrew/legacy-homebrew/pull/42845 ) prior to moving to Homebrew.

_side note:_ Due to some changes in my employment, I had not been able to devote almost any time to Homebrew over the past year. However, this has markedly changed recently and I would like to continue with the move of the OSGeo formulae to `homebrew-geospatial`. I am starting to clean up our tap in preparation for such a move and will be soliciting more devs to help with initial formulae hosted there.

**Pull request closed without merging**